terraform-tests/redis_test.go

```diff
@@ -266,6 +272,143 @@ var _ = Describe("Redis", Label("redis-terraform"), Ordered, func() {
+					"destination_type": Equal("cloudwatch-logs"),
+					"log_format":       Equal("json"),
+					"log_type":         Equal("slow-log"),
+				})),
+			}))
+		})
+
+		Context("slow_log loggroup configuration", func() {
+			kmsKeyId := ""
+			retentionInDays := 0
+			JustBeforeEach(func() {
+				plan = ShowPlan(terraformProvisionDir, buildVars(defaultVars, map[string]any{
+					"logs_slow_log_loggroup_kms_key_id":        kmsKeyId,
+					"logs_slow_log_loggroup_retention_in_days": retentionInDays,
+					"logs_slow_log_enabled":                    true,
+				}))
+			})
+
+			It("should configure the log group with default values", func() {
+				slowLogResource := "aws_cloudwatch_log_group"
+				Expect(AfterValuesForType(plan, slowLogResource)).To(MatchKeys(IgnoreExtras, Keys{
+					"name":              Equal("/aws/elasticache/cluster/csb-redis-test/slow-log"),
+					"retention_in_days": BeNumerically("==", retentionInDays),
+					"kms_key_id":        BeNil(),
+					"skip_destroy":      BeFalse(),
+					"tags":              HaveKeyWithValue("key1", "some-redis-value"),
+				}))
+			})
+
+			Context("providing explicit values", func() {
+				BeforeEach(func() {
+					kmsKeyId = "test-kms-key"
+					retentionInDays = 180
+				})
+
+				It("should configure the passed values", func() {
+					slowLogResource := "aws_cloudwatch_log_group"
+					Expect(AfterValuesForType(plan, slowLogResource)).To(MatchKeys(IgnoreExtras, Keys{
+						"name":              Equal("/aws/elasticache/cluster/csb-redis-test/slow-log"),
+						"retention_in_days": BeNumerically("==", retentionInDays),
+						"kms_key_id":        Equal(kmsKeyId),
+						"skip_destroy":      BeFalse(),
+						"tags":              HaveKeyWithValue("key1", "some-redis-value"),
+					}))
+				})
+			})
+		})
+
+	})
+
+	Context("engine_log is enabled", func() {
+		BeforeAll(func() {
+			plan = ShowPlan(terraformProvisionDir, buildVars(defaultVars, map[string]any{
+				"logs_engine_log_loggroup_kms_key_id":        "",
+				"logs_engine_log_loggroup_retention_in_days": 0,
+				"logs_engine_log_enabled":                    true,
+			}))
+		})
+
+		It("should create a log group for engine log", func() {
+			expectedResources := []string{"aws_cloudwatch_log_group"}
+			expectedResources = append(expectedResources, getExpectedResources()...)
+			Expect(ResourceChangesTypes(plan)).To(ConsistOf(expectedResources))
+		})
+
+		It("should assign that log group to the replication group", func() {
+			Expect(AfterValuesForType(plan, resource)).To(MatchKeys(IgnoreExtras, Keys{
+				"log_delivery_configuration": ConsistOf(MatchAllKeys(Keys{
+					"destination":      Equal("/aws/elasticache/cluster/csb-redis-test/engine-log"),
+					"destination_type": Equal("cloudwatch-logs"),
+					"log_format":       Equal("json"),
+					"log_type":         Equal("engine-log"),
+				})),
+			}))
+		})
+
+		Context("engine_log loggroup configuration", func() {
+			kmsKeyId := ""
```

Comment:
Because this is an ordered test, setting values outside of a `BeforeEach()` works, but I'd strongly recommend against it because the test starting on line 381 has:
```go
"retention_in_days": BeNumerically("==", retentionInDays),
```

If another piece of code were to update the value of `retentionInDays` then the test would continue to pass, but it would not be testing what you think it's testing.

Reply:
Thanks, fixing this